src/seedu/addressbook/Main.java

```diff
@@ -111,12 +113,17 @@ private CommandResult executeCommand(Command command)  {
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e){
+            ui.showReadOnlyError();
```

Comment:
Does StorageOperationException only catch read-only errors? There could also be file in use exceptions.